02-microarray/dimension-reduction_microarray_02_umap.Rmd

```diff
@@ -0,0 +1,321 @@
+
+```{r}
+if (!("umap" %in% installed.packages())) {
+  # Install umap package
+  BiocManager::install("umap", update = FALSE)
+}
+```
+
+Attach the packages we need for this analysis:
+
+```{r}
+# Attach the `umap` library
+library(umap)
+
+# Attach the `ggplot2` library
+library(ggplot2)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+
+# Set the seed so our results are reproducible:
+set.seed(12345)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE37382.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE37382.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+
+Now we are going to use a combination of functions from the `umap` and the `ggplot2` packages to perform and visualize the results of the Uniform Manifold Approximation (UMAP) dimension reduction technique on our medulloblastoma samples.
+
+## Perform Uniform Manifold Approximation (UMAP)
+
+In this code chunk, we are going to perform Uniform Manifold Approximation (UMAP) on our data and create a data frame using the UMAP scores and the variables from our metadata that we are going to use to annotate our plot later.
+
+```{r}
+# Perform Uniform Manifold Approximation (UMAP)
+umap_data <- umap::umap(t(df)) # We have to transpose our data frame to meet the `umap::umap()` function requirements
+
+# Make into data frame for plotting with `ggplot2`
+umap_df <- data.frame(umap_data$layout) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame; match by sample ids
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Now, let's plot the UMAP scores.
+
+```{r}
+# Make a scatterplot using `ggplot2` functionality
+umap_plot <- ggplot(
+  umap_df,
+  aes(
+    x = X1,
+    y = X2
+  )
+) +
+  geom_point() + # This tells R that we want a scatterplot
+  theme_classic() # This tells R to return a classic-looking plot with no gridlines
+
+# Print out plot here
+umap_plot
+```
+
```

Comment:
Can we add a sentence that previews the next steps. Something like what we have elsewhere `It's hard to interpret our UMAP results without some metadata labels` https://github.com/AlexsLemonade/refinebio-examples/blob/7ab7fafbf3cbe171841cb4f5eac5184c123189a0/03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd#L344